### PR TITLE
add try/catch to handle promise errors

### LINKED ISSUE #8: exception not handled in checkForPlayStationRedirect async await
(node:76664) UnhandledPromiseRejectionWarning: Error: connect ETIMEDOUT 69.192.7.202:443
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1129:14)
(node:76664) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:76664) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

I will follow up with a PR later today.

### LINKED ISSUE #19: Error after 500 retries.
image

got this error after 500 or so retries.

### LINKED ISSUE #25: After 2632 tries, TimeoutError occured
Windows 10, fresh Node install. Running through Windows Terminal.

Everything runs and the test commands work great, but I left the script running overnight and when I woke up the next morning I found this message. 
Screenshot 2020-12-11 140546
There are a few warnings, but most obvious one to me is "TimeoutError: Navigation timeout of 30000 ms exceeded"

I wish I had more insight into what caused this, but it seems like an important issue if this script is going to accomplish what it sets out to do. I'll be leaving it on overnight again and will report back if I can replicate it.

### LINKED ISSUE #38: Timing Out, Won't run on its own
Running on Windows 10 from Documents folder. Node was installed into my program files.

I am getting a timeout error that precents the bot from running on its own for long periods of time.  Highest number of tries i got was 297.  This has been pretty consistent at failing around the 30 try mark.  Below is the full dump of error message:

C:\Users\thund\Documents\PS5\get-my-ps5-master\get-my-ps5-master\node_modules\puppeteer\lib\cjs\puppeteer\common\LifecycleWatcher.js:106
        return new Promise((fulfill) => (this._maximumTimer = setTimeout(fulfill, this._timeout))).then(() => new Errors_js_1.TimeoutError(errorMessage));
                                                                                                              ^

TimeoutError: Navigation timeout of 30000 ms exceeded
    at C:\Users\thund\Documents\PS5\get-my-ps5-master\get-my-ps5-master\node_modules\puppeteer\lib\cjs\puppeteer\common\LifecycleWatcher.js:106:111
npm ERR! code 1
npm ERR! path C:\Users\thund\Documents\PS5\get-my-ps5-master\get-my-ps5-master
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node index

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\thund\AppData\Local\npm-cache\_logs\2020-12-12T17_39_33_233Z-debug.log

### PR DESCRIPTION
**Pull request by outphase84**
added error hardening for failures to retrieve page

### CODE REVIEW

**Comment by camtheperson** [comment]
@outphase84 Please update your target branch from `camtheperson:master` to `camtheperson:rc/1.2.0` (https://github.com/camtheperson/get-my-ps5/tree/rc/1.2.0).

File: utils.js [hunk 1]
```diff
@@ -38,14 +38,21 @@ function addToCartLoop(id, guid, numTries, checkInterval = 10000) {
  * @param onSuccess - Callback function for successful redirect
  */
 async function checkForPlaystationDirectRedirect(checkInterval, onSuccess, version, browser, numTries = 1) {
+    // Declare these variables ahead of time in case of error catch
+    var response;
+    var responseBody;
+    var responseStatus;
```
Comment (camtheperson) [inline utils.js]: Change to `let` instead of using `var` here.

**Review by camtheperson — CHANGES_REQUESTED** [review]

File: utils.js (hunk 1 above)
Comment (camtheperson) [reply utils.js]: You can also chain them, like:
```
let response, responseBody, responseStatus;
```

**Comment by outphase84** [comment]
Replaced function scope decs with block scope.  Kept explicit declaration instead of chained, my OCD doesn't like chaining declarations

**Review by camtheperson — APPROVED** [review]

**Pull request merged**